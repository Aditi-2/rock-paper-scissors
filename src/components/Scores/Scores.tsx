import React from "react";
import { useSelector } from "react-redux";
import styled from "@emotion/styled";
import { RootState } from "../../redux/rootReducer";
import { GameScrore, GameType } from "../../common/types/game";

interface ScoresProps {
  scores: GameScrore;
  gameType: GameType;
}

export const Scores: React.FC<ScoresProps> = ({scores, gameType}) => {
  return (
      <ScoresStyled>
        <div className="playerOne">
          {gameType === "PLAYER_VS_COMPUTER" ? (
            <p>You</p>
          ) : (
            <p>Computer One</p>
          )}
          <PlayerScoreStyled>{scores.playerOne}</PlayerScoreStyled>
        </div>
        <div className="playerTwo">
          {gameType === "PLAYER_VS_COMPUTER" ? (
            <p>Computer</p>
          ) : (
            <p>Computer two</p>
          )}
          <PlayerScoreStyled>{scores.playerTwo}</PlayerScoreStyled>
        </div>
      </ScoresStyled>
  );
};

const ScoresStyled = styled.div`
  background-color: #dedede;
  color: #141942;
  border: 1px solid #000;
  border-radius: 5px;
  width: 38%;
  display: flex;
  justify-content: space-around;
  margin: 0 auto;
  font-size: 18px;
  padding: 1rem;
`;

const PlayerScoreStyled = styled.span`
  font-size: 20px;
  font-weight: bold;
`;
