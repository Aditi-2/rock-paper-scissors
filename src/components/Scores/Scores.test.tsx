import React from 'react';
import {create} from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Scores } from './Scores';

const getComponent = () => <Scores gameType="PLAYER_VS_COMPUTER" scores={{playerOne: 2, playerTwo: 3}}/>

test('Scores Component snapshot', () => {
    const component = create(getComponent()).toJSON();
    expect(component).toMatchSnapshot();
});

test('renders Scores Component', () => {
  render(getComponent());
  const playerText = screen.getByText(/You/i);
  expect(playerText).toBeInTheDocument();
});

