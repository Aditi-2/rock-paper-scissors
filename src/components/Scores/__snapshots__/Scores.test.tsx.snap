// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scores Component snapshot 1`] = `
<div
  className="css-9i3zow"
>
  <div
    className="playerOne"
  >
    <p>
      You
    </p>
    <span
      className="css-1aylpg"
    >
      2
    </span>
  </div>
  <div
    className="playerTwo"
  >
    <p>
      Computer
    </p>
    <span
      className="css-1aylpg"
    >
      3
    </span>
  </div>
</div>
`;
